client/src/connection_cache.rs

```diff
@@ -251,7 +269,7 @@ impl ConnectionPool {
 }
 
 impl ConnectionCache {
-    pub fn new(use_quic: bool, connection_pool_size: usize) -> Self {
+    pub fn new(use_quic: UseQUIC, connection_pool_size: usize) -> Self {
```

Comment:
`ConnectionCache` is something we expect users of solana-client to be aware of right? 

If so, then I think it's better for `ConnectionCache::new()` to imply QUIC and there's a different constructor, `ConnectionCache::with_tpu()`, for the TPU case that can be later deprecated and removed. Otherwise we're stuck with that first `use_quic` arg even well after QUIC is enabled.

Reply:
I am hoping that we can remove the `use_quic` flag (and `UDP` path) altogether once we enable QUIC completely.

Reply:
Create #26117